tests/instant_articles/test_facebook_api.py

```diff
@@ -0,0 +1,149 @@
+import requests_mock
+
+from django.utils import timezone
+from django.test.utils import override_settings
+
+from bulbs.content.models import FeatureType, Content
+from bulbs.utils.test import BaseIndexableTestCase
+from bulbs.utils.test.mock_vault import mock_vault
+
+from example.testcontent.models import TestContentObjThree
+
+
+class FacebookAPITestCase(BaseIndexableTestCase):
+
+    def setUp(self):
+        super(FacebookAPITestCase, self).setUp()
+        self.ft = FeatureType.objects.create(name="NIP", instant_article=True)
+
+    @override_settings(
+        FACEBOOK_PAGE_ID='123456',
+        FACEBOOK_API_BASE_URL='https://graph.facebook.com/v2.6',
+        FACEBOOK_API_ENV='production',
+        FACEBOOK_TOKEN_VAULT_PATH='facebook/onion_token',
+        WWW_URL='www.theonion.com',
+        BETTY_FIXED_URL='http://i.onionstatic.com/onion'
+    )
+    @mock_vault({'facebook/onion_token': 'TOKEN'})
+    def test_publish(self):
+        with requests_mock.mock() as mocker:
+            # post to instant article endpoit
+            mocker.post(
+                "https://graph.facebook.com/v2.6/123456/instant_articles",
+                status_code=200,
+                json={
+                    "id": 456
+                })
+            # get status of instant article
+            mocker.get(
+                "https://graph.facebook.com/v2.6/456?access_token=TOKEN",
+                status_code=200,
+                json={
+                    "id": 9876,
+                    "status": "SUCCESS"
+                })
+
+            content = TestContentObjThree.objects.create(
+                body="<p>This is the body</p>",
+                feature_type=self.ft)
+            content.published = timezone.now()
+            content.save()
+
+            Content.search_objects.refresh()
+            c = Content.objects.get(id=content.id)
+
+            # check that the ia_id is set & call count is correct
+            self.assertEqual(c.instant_article_id, 9876)
+            self.assertEqual(mocker.call_count, 2)
+
+    @override_settings(
```

Comment:
class decorators weren't working for some reason (python 2 support?), but i was able to pair these down, so it's not as large.